src/Tearing/TearingScenarioEngine.inl

```diff
@@ -0,0 +1,176 @@
+    {
+
+        
+        addInput(&d_startVertexId);
+        addInput(&d_startTriId);
+        addInput(&d_startDirection);
+        addInput(&d_startLength);
+    }
+
+    template <class DataTypes>
+    void TearingScenarioEngine<DataTypes>::init()
+    {
+        BaseTearingEngine<DataTypes>::init();
+
+    }
+
+    template <class DataTypes>
+    void TearingScenarioEngine<DataTypes>::reinit()
+    {
+        BaseTearingEngine<DataTypes>::update();
+    }
+
+    template <class DataTypes>
+    void TearingScenarioEngine<DataTypes>::handleEvent(sofa::core::objectmodel::Event* event)
+    {
+        BaseTearingEngine<DataTypes>::handleEvent(event);
+    }
+
+
+    // --------------------------------------------------------------------------------------
+    // --- Computation methods
+    // --------------------------------------------------------------------------------------
+    template<class DataTypes>
+    inline void TearingScenarioEngine<DataTypes>::computePerpendicular(Coord dir, Coord& normal)
+    {
+        int  triID = d_startTriId.getValue();
+        int indexA = d_startVertexId.getValue();
+
+        if (triID == -1) 
+            return;
+       
+        // Access topology 
+        sofa::core::topology::BaseMeshTopology* topo = this->getTopology();
+        const Triangle& VertexIndicies = topo->getTriangle(triID);
+        constexpr size_t numVertices = 3;
+       
+        Index B_id = -1, C_id = -1;
+
+        for (unsigned int vertex_id = 0; vertex_id < numVertices; vertex_id++)
+        {
+            if (VertexIndicies[vertex_id] == indexA)
+            {
+                B_id = VertexIndicies[(vertex_id + 1) % 3];
+                C_id = VertexIndicies[(vertex_id + 2) % 3];
+                break;
+            }
+        }
+
+        helper::ReadAccessor< Data<VecCoord> > x(d_input_positions);
+        Coord A = x[triID];
+        Coord B = x[B_id];
+        Coord C = x[C_id];
+
+        Coord AB = B - A;
+        Coord AC = C - A;
+
+        Coord triangleNormal = sofa::type::cross(AB, AC);
+        normal = sofa::type::cross(triangleNormal, dir);
+
+    }
+
+
+
+    template <class DataTypes>
+    void TearingScenarioEngine<DataTypes>::algoFracturePath()
+    {
+       
+        int indexA = d_startVertexId.getValue();
+        int triID = d_startTriId.getValue();
+        const Vec3& dir = d_startDirection.getValue();
+        const Real& alpha = d_startLength.getValue();
+
+        helper::ReadAccessor< Data<VecCoord> > x(d_input_positions);
+        Coord Pa = x[indexA];
+        
+        
+        Coord normal;
+        computePerpendicular(dir,normal);
+
+
+        Coord Pb = Pa + alpha * dir;
+        Coord Pc = Pa - alpha * dir;
+
+        TearingAlgorithms<DataTypes>* tearingAlgo = this->getTearingAlgo();
+        if (tearingAlgo == nullptr)
+            return;
+        
+        tearingAlgo->algoFracturePath(Pa, indexA, Pb, Pc, triID, normal, d_input_positions.getValue());
```

Comment:
```suggestion
        tearingAlgo->algoFracturePath(Pa, indexA, Pb, Pc, triID, normal, this->d_input_positions.getValue());
```

When you are using member from the Base class, you need to add this-> because it is a templated class